lib/src/parse/stylesheet.dart

```diff
@@ -2269,7 +2269,9 @@ abstract class StylesheetParser extends Parser {
         case $cr:
         case $lf:
         case $ff:
-          if (indented) break loop;
+          if (indented && peekIndentationOfNextLine() > currentIndentation)
```

Comment:
I don't think this logic is right. The indented syntax has very specific logic for multiline selectors, which isn't just "fold together everything on adjacent lines with equal or lower indent". The next line should be consumed as a selector if

1. the current line ended with a comma (possibly followed by whitespace and comments), and
2. the next line has the same level of indentation.

This is going to be kind of tricky to implement here, since we consume selectors as `_almostAnyValue()` rather than just going line-by-line like Ruby's parser does.

Reply:
Yes, I see this logic doesn't seem right. But I thought `_almostAnyValue` should do the same for sass and scss.

After `_almostAnyValue` is called in _styleRule it assigns the value directly to the selector and then continues with the children. For sass and scss the parsed selector are different.

For scss
```dart
    var selector = _almostAnyValue(); // selector: a,
    var children = this.children(_statement);
```
For sass
```dart
    var selector = _almostAnyValue(); // selector: a,\nb
    var children = this.children(_statement);
```

So I think it should be corrected in `_almostAnyValue`

Reply:
I think the best way to handle this is to have `_almostAnyValue()` break after *any* newline in Sass mode, and override `styleRule()` in `SassParser` to check for trailing commas and consume additional lines as necessary.